From 5926a2cb526f11d4a4a3cd3980e5894f8e96e615 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Thu, 12 Jan 2012 10:23:38 +0100
Subject: [PATCH] 98usrmount: don't pass fstab fs options as fsck options

For example for line in /etc/fstab:

  /dev/vg/usr  /usr  ext4  nosuid,nodev  0 2

fsck_single is called as follows:

  fsck_single /dev/vg/usr ext4 nosuid,nodev

and this eventually resolves to:

  e2fsck -a nosuid,nodev /dev/vg/usr

which causes e2fsck usage error.
---
 modules.d/98usrmount/mount-usr.sh |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/98usrmount/mount-usr.sh b/modules.d/98usrmount/mount-usr.sh
index a61f935..fc205d5 100755
--- a/modules.d/98usrmount/mount-usr.sh
+++ b/modules.d/98usrmount/mount-usr.sh
@@ -19,7 +19,7 @@ mount_usr()
 
     if [ "x$_usr_found" != "x" ]; then
         # we have to mount /usr
-        fsck_single "$_dev" "$_fs" "$_opts"
+        fsck_single "$_dev" "$_fs"
         _ret=$?
         echo $_ret >/run/initramfs/usr-fsck
         if [ $_ret -ne 255 ]; then
@@ -29,4 +29,4 @@ mount_usr()
     fi
 }
 
-mount_usr
\ No newline at end of file
+mount_usr
-- 
1.7.8.3

